words/core.asm

```diff
@@ -5028,10 +5028,15 @@ w_s_quote:
                 stz tmp2+1
 
 s_quote_start:
-                ; Put a jmp past the string data to be filled in later.
-                jsr cmpl_jump_later
-                jsr w_here             ; the start of the string
-                ; ( jmp-target addr )
+                ; write the string to unallocated space beyond PAD
+                ; by temporarily incrementing cp by $200
+                ;TODO is possible OoM worse than losing space for every interactive string?
```

Comment:
this is the unsafe interpreted string, which gets compiled by w_sliteral below if compiled.
could also revert to previous approach where any string uses up dictionary space as an unnamed sliteral

Reply:
This was one of those decisions that was made a while ago that can certainly be revisited.  It was decided, at the time, that it was better to just save the strings into the dictionary.  This was to make it easier to implement >255 length strings (which Tali did not support prior).  Your proposed method does have the possibility of running out of room when memory is tight, but has the benefit of using less dictionary space for strings during regular use.  One big advantage of compiling all strings into the dictionary is that they all stay where you put them and won't be overwritten, even if you perform other operations.  Here is a possible fail using the method you propose, for example:
```
s" This is a long string" 
\ The "addr u" from above stack stays on stack until the MOVE at the end
#1024 buffer: mybuffer
mybuffer #1024 blank \ This overwrites what used to be at cp+$200
mybuffer swap move \ This should copy the string into the buffer
```
This currently works as expected, but I believe will have only spaces in mybuffer on your proposed solution.

It's worth noting that `S"` in the CORE word set has undefined interpretation semantics, but the the version in the FILE wordset does have interpretation semantics (see [https://forth-standard.org/standard/file/Sq](https://forth-standard.org/standard/file/Sq) for the details).  The important part is:

> Since an implementation may choose to provide only one buffer for interpreted strings, an interpreted string is subject to being overwritten by the next execution of S" in interpretation state. It is intended that **no standard words other than S" should in themselves cause the interpreted string to be overwritten**. However, since words such as [EVALUATE](https://forth-standard.org/standard/core/EVALUATE), [LOAD](https://forth-standard.org/standard/block/LOAD), [INCLUDE-FILE](https://forth-standard.org/standard/file/INCLUDE-FILE) and [INCLUDED](https://forth-standard.org/standard/file/INCLUDED) can result in the interpretation of arbitrary text, possibly including instances of S", the interpreted string may be invalidated by some uses of these words.

and this is why Tali currently compiles all strings into the dictionary.  Otherwise, we would need to reserve a buffer of 80 characters (that's the minimum allowed in the FILE wordset) somewhere that is not the dictionary (probably just before the dictionary, like the block buffer currently is done, or at the end of RAM, like the command line history is done).

For those using the CORE wordset, their ANS-compliant code should not be using `S"` in interpreted mode anyway, so the decision may not matter that much.  When you updated the tests, you technically made the tests more ANS compliant and less Tali-specific.

I don't think the cp+xxx method will work here.  I think the two main options are to reserve a separate buffer of at least 80 characters somewhere specifically for `S"` and `S\"` to use (I'm not opposed), or to compile all strings into the dictionary.  Compiling into the dictionary has the benefit that they can be longer than a pre-set buffer (>80 or >255), that they are already there when they actually _need_ to be compiled into the dictionary, and the only difference is whether to return the `addr u` immediately on the stack (interpreted mode) or do an SLITERAL runtime in front (compiled mode).